### PR TITLE
Thread sanitizer tests in the CI pipeline

### PR DESCRIPTION
**Pull request by jblueh**
## Proposed Changes
To simplify the maintenance of hybrid parallel SU2, we want to run thread sanitizer tests as part of the CI pipeline. We use thread-sanitizer enabled containers for this. This is work in progress and there are things that need testing, so I want to take it step by step while observing the behaviour of the CI pipeline.

## Related Work
https://github.com/su2code/Docker-Builds/pull/17 and follow-on PRs, #1679

## PR Checklist

- [X] I am submitting my contribution to the develop branch.
- [x] My contribution generates no new compiler warnings (try with --warnlevel=3 when using meson).
- [x] My contribution is commented and consistent with SU2 style (https://su2code.github.io/docs_v7/Style-Guide/).
- [x] I have added a test case that demonstrates my contribution, if necessary.
- [ ] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp), if necessary.

### CODE REVIEW

File: .github/workflows/regression.yml [hunk 1]
```diff
@@ -68,7 +68,50 @@ jobs:
...
+    if: inputs.runner != 'ARM64'
+    strategy:
+      fail-fast: false
+      matrix:
+        config_set: [BaseOMP-tsan, ReverseOMP-tsan] #ForwardOMP-tsan
+        include:
+          - config_set: BaseOMP-tsan
+            flags: '--buildtype=debugoptimized -Dwith-omp=true -Denable-mixedprec=true -Denable-tecio=false --warnlevel=3'
+          - config_set: ReverseOMP-tsan
+            flags: '--buildtype=debugoptimized -Denable-autodiff=true -Denable-normal=false -Dwith-omp=true -Denable-mixedprec=true -Denable-tecio=false --warnlevel=3'
```
Comment (jblueh) [inline .github/workflows/regression.yml:86]: Optimized debug builds have reasonable runtime and stack traces that are still useful if a data race is detected. This is in line with the [recommendations](https://github.com/google/sanitizers/wiki/ThreadSanitizerCppManual).

I had to remove the `-werror`, there is warning about control reaching the end of the non-void function `CMultizoneDriver::Monitor()`. IIRC a fix for this is pending somewhere?

**Review by jblueh — COMMENTED** [review]
Thread sanitizer tests are now executed as part of the CI pipeline. With the tweaks explained in the code comments, the time spent on thread sanitizer builds and tests is right now approximately
| job | time |
|---------------------------------|-----------|
| build `BaseOMP-tsan` | 20min |
| build `ReverseOMP-tsan` | 45min |
| run `hybrid_regression.py` with tsan | 42min |
| run `hybrid_regression_AD.py` with tsan | 12min |

which is reasonable, I would say, given that thread sanitizer runs are expected to be slower and that some of the other build and test jobs take around 20min as well.

I will update the documentation on containers with thread sanitizer specifics when this is merged.

File: .github/workflows/regression.yml (hunk 1 above)
Comment (pcarruscag) [reply .github/workflows/regression.yml:86]: Yep in #2011

File: .github/workflows/regression.yml [hunk 2]
```diff
@@ -52,12 +52,12 @@ jobs:
           key: ${{ matrix.config_set }}-${{ github.sha }}
           restore-keys: ${{ matrix.config_set }}
       - name: Pre Cleanup
-        uses: docker://ghcr.io/su2code/su2/build-su2:230225-2136
```
Comment (pcarruscag) [inline .github/workflows/regression.yml:55]: Update the release_management action as well to use the new images please

File: .github/workflows/regression.yml (hunk 1 above)
Comment (jblueh) [reply .github/workflows/regression.yml:86]: The `-werror` can be readded there :+1:

File: .github/workflows/regression.yml (hunk 2 above)
Comment (jblueh) [reply .github/workflows/regression.yml:55]: Updated in c6931d7714f2b51dc699ce4bb0b53ef83e8674c7

**Pull request merged**